### ISSUE TITLE
Component with inner element doesn't render

### ISSUE THREAD

**Issue by JanMiksovsky** [webcomponents/webcomponentsjs#862]
See http://jsbin.com/zixupiz/edit?html,output.

The repro case defines a trivial outer-element whose shadow contains an inner-element and a slot. This renders in Chrome/Safari, but doesn't render in the polyfill. (Open the repro case above in Firefox or Edge. The bug also shows up in IE if the repro's ES6 is transpiled to ES5.)

One odd symptom: the light DOM content put inside the outer-element can't even be seen in the DOM — like it's been removed entirely.

The order of the inner-element in the outer's shadow root matters. If the inner-element is moved to the end of the shadow, the component renders as expected.

**Comment by azakus** [comment]
Slightly adapted to force ShadyDOM in Chrome: http://jsbin.com/xirevok/1/edit?html,output

**Comment by sorvell** [comment]
The ShadyDOM polyfill uses an inert element to parse `innerHTML` text into elements before moving them into the containing element. In this case the inner-element's call to `innerHTML` occurs *during* the set of the outer-element's `innerHTML` and inert element's content is stomped on. Keeping distinct inert elements is an easy and relatively cheap fix.

**Comment by sorvell** [comment]
According to spec, the custom element should upgrade only after the `innerHTML` is complete; however, we cannot guarantee this behavior here and thus need the above fix.

In conjunction with a fix for this [issue](https://github.com/webcomponents/custom-elements/issues/73) in custom-elements, we can be more correct when custom elements are polyfilled by using a captured appendChild here.

However, the fix described above will be needed when native custom elements are used since we simulating `innerHTML` using `appendChild` cannot be 100% spec compliant with respect to processing of `innerHTML`.

**Comment by JanMiksovsky** [comment]
I notice that the discussion here has focused on `innerHTML`, and just wanted to note that I'd used `innerHTML` in this repro only because it was compact. In the actual code in which we first encountered this bug, we're actually populating the shadow in a more conventional way: cloning a template, then calling `appendChild` to add the cloned tree to the shadow.

If the fix for this bug would cover that too, then we're fine, but if the current fix is specific to `innerHTML`, there's a chance our original bug would still exist. If you'd be interested in seeing a more representative repro case of the code we're using in practice, please let me know.

**Comment by justinfagnani** [comment]
@JanMiksovsky sounds to me like that would definitely be useful

**Comment by JanMiksovsky** [comment]
Okay. I created a newer repro case that uses a template and `appendChild` in a very conventional way: http://jsbin.com/jeqaram/edit?html,output

This generally works, but fails on a very recent IE 11 (11.15.16299.0).

It looks like this is blocking some of our components from working in IE, which is what caused me to file the original bug. In isolating the earlier repro, it's possible I stumbled on a separate bug related specifically to `innerHTML`, whereas this may be unique to IE.

**Comment by azakus** [comment]
@JanMiksovsky the `appendChild` case is a separate issue, which I've filed as webcomponents/shadydom#196

**Comment by JanMiksovsky** [comment]
Running @azakus' modified jsbin above with v1.0.20, this appears to be fixed. As far as I can tell, this can be closed.

**Comment by justinfagnani** [comment]
Awesome!
